mirrord-agent/src/main.rs

```diff
@@ -183,10 +183,20 @@ impl ClientConnectionHandler {
         id: ClientID,
         stream: TcpStream,
         pid: Option<u64>,
+        ephemeral: bool,
         sniffer_command_sender: Sender<SnifferCommand>,
         cancel_token: CancellationToken,
     ) -> Result<(), AgentError> {
-        let file_manager = FileManager::new(pid);
+        let file_manager = match pid {
+            Some(_) => FileManager::new(pid),
+            None => {
```

Comment:
I think this can be formatted better (https://doc.rust-lang.org/book/ch18-03-pattern-syntax.html)
`None if ephemeral`
`None if !ephemeral`